src/main/java/io/pravega/connectors/flink/AbstractStreamingWriterBuilder.java

```diff
@@ -23,9 +23,9 @@
  */
 public abstract class AbstractStreamingWriterBuilder<T, B extends AbstractStreamingWriterBuilder> extends AbstractWriterBuilder<B> {
 
-    // the numbers below are picked somewhat arbitrarily at this point
-    private static final long DEFAULT_TXN_TIMEOUT_MILLIS = 2 * 60 * 60 * 1000; // 2 hours
-    private static final long DEFAULT_TX_SCALE_GRACE_MILLIS = 10 * 60 * 1000; // 10 minutes
+    // the numbers below are picked based on the default max settings in Pravega
+    private static final long DEFAULT_TXN_TIMEOUT_MILLIS = 30000; // 30 seconds
```

Comment:
This timeout value corresponds to the lease period of a transaction. The writer renews the lease within the configured period, and it does it automatically, without any necessary intervention from the application. 30s sounds like a pretty reasonable default value to me.

This configuration parameter also determines the maximum amount of time we allow a txn to remain open. The maximum execution time is calculated as the minimum of 1 day and 1000x the lease period.